File: leetcode/lc-us/cpp/26.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int> &nums) {
        int low = 0, count = 0, prev = INT32_MIN;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != prev) {
                nums[low] = nums[i];
                low++;
                count++;
            }
            prev = nums[i];
        }
        return count;
    }
};

File: leetcode/lc-us/py/26.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        i = j = 0
        while j < n:
            nums[i] = nums[j]
            i += 1
            k = j + 1
            while k < n and nums[k] == nums[j]:
                k += 1
            j = k
        return i
